### DISCUSSION TITLE
How to use running environment isolation in ScriptX ?

### DISCUSSION THREAD

**Discussion by yqs112358** [Tencent/ScriptX#19] [Q&A] [answered]
Use multiple engines？Will it affect efficiency and perfomance?

**Comment by LanderlYoung** [comment, marked as answer]
Yes, it's the EnigneScope. refer to docs here https://github.com/Tencent/ScriptX/blob/main/docs/zh/Basics.md#enginescope-%E4%B8%8E-stackframescope and/or here https://github.com/Tencent/ScriptX/blob/main/docs/en/Basics.md#enginescope-and-stackframescope

**Reply by yqs112358** [reply]
Thanks a lot !
However, I have tried code below
![image](https://user-images.githubusercontent.com/37969157/115171178-15e9af00-a0f5-11eb-93c7-3be2d3d82a1d.png)
It's strange that the last line still outputs that aaa=5
Can it means so-called running environment isolation?

**Reply by yqs112358** [reply]
> 通常来说你不需要自己使用一个StackFrameScope，因为EngineScope已经蕴含了一个（这样你才能使用Local引用）

However, it works like they are in the same StackFrameScope.
Is it means that it still needs different engines to isolate the environment?
@LanderlYoung

**Reply by LanderlYoung** [reply]
No, you misunderstood the concept.

The second `EngineScope` enters the same engine (the variable `engine`), so there is no difference.

maybe try this:

```c++
EngineScope(engine1);
engine1->eval ...
{
EngineScope(engine2);
engine2->eval ...
}
print log

```

**Reply by yqs112358** [reply]
Thanks very much
It works. 😉
